examples/roll.py

```diff
@@ -0,0 +1,75 @@
+"""An example of a simple CLI for rolling dice.
+
+The example demonstrates the usage of a typed argument
+accepting a list of custom objects (instances of `Die`).
+
+Usage:
+    $ python roll.py 1d6
+    3
+    $ python roll.py 2d6
+    2 5
+    $ python roll.py 1d6 2d3
+    6
+    2 1
+"""
+
+import re
+import random
+from typing import List
+
+from candies.cli import cli, Arg
+
+
+class Die:
+    """A die to roll."""
+
+    def __init__(self, sides: int, rolls: int):
+        self._sides = sides
+        self._rolls = rolls
+
+    def roll(self) -> List[int]:
+        """Rolls the die the required amount of times."""
+        return [random.randint(1, self._sides) for _ in range(self._rolls)]
+
+
+@cli
+def roll(dice: Arg[List[Die]]('A list of dice to roll.')):
+    """Roll dice.
+
+    Each die should be specified using the dice notation: [m]d{n}.
```

Comment:
I think `using the following dice notation` would be a more accurate statement.

Reply:
What is the difference?

Reply:
I'm not sure, but for me `using the dice notation` sounds like a reference to the notion of [_Dice notation_](https://en.wikipedia.org/wiki/Dice_notation). But in the example we don't actually support the _full_ notation, so saying `the following` makes a sentence a bit more accurate.

Maybe `using the following notation` would be even better.

This is a totally minor stuff, so you don't need to change anything if you think there is no difference.

Reply:
Ah, _now_ I see. As I understand, `[m]d{n}` is the default notation, and `[m]d{n} + k` and others are extensions.